samples/shields/npm6001_ek/src/main.c

```diff
@@ -190,6 +424,25 @@ static int cmd_wdt_kick(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_npm6001_regulator_cmds,
```

Comment:
This feels a bit like a reimplementation of the regulator shell. Is there any reason that the regulator shell couldn't be used here? It looks like it would need extensions to list all supported voltages and report error conditions, but other than that it seems that the use cases of this shell implementation would be covered.

Reply:
Yup, will work on improving shell experience soon, see my comment here for the reason: https://github.com/zephyrproject-rtos/zephyr/pull/52358#discussion_r1028165960

Reply:
Ok, makes sense